core/index/index_writer.cpp

```diff
@@ -2111,9 +2119,10 @@ index_writer::flush_pending(flush_context& ctx,
   return {std::move(segment_flush_locks), max_tick};
 }
 
-index_writer::pending_context_t index_writer::flush_all() {
+index_writer::pending_context_t index_writer::flush_all(progress_report_callback const& progress) {
   REGISTER_TIMER_DETAILED();
 
+  assert(progress);
```

Comment:
since both `start()` and `commit()` accept `progress_report_callback`, can we move emptiness check to `flush_all`